spec/integration/synthetics_spec.rb

```diff
@@ -0,0 +1,131 @@
+require_relative '../spec_helper'
+
+describe Dogapi::Client do
+  SYNTHETICS_TEST_PUBLIC_ID = '84r-szk-xpt'
+  SYNTHETICS_TESTS_PUBLIC_IDS = ['84r-szk-xpt', 'sti-s2m-ciz']
+  NEW_STATUS = 'paused'
+  SYNTHETICS_RESULT_ID = 123_456
+
+  SYNTHETICS_TEST_TYPE = 'api'.freeze
+  SYNTHETICS_TEST_CONFIG = {
+    'assertions' => [
+      {
+        'operator' => 'is',
+        'type' => 'statusCode',
+        'target' => 403
+      },
+      {
+        'operator' => 'is',
+        'property' => 'content-type',
+        'type' => 'header',
+        'target' => 'text/html'
+      },
+      {
+        'operator' => 'lessThan',
+        'type' => 'responseTime',
+        'target' => 2000
+      }
+    ],
+    'request' => {
+      'method' => 'GET',
+      'url' => 'https://datadoghq.com',
+      'timeout' => 30,
+      'headers' => {
+        'header1' => 'value1',
+        'header2' => 'value2'
+      },
+      'body' => 'body to send with the request'
+    }
+  }.freeze
+
+  SYNTHETICS_TEST_OPTIONS = {
+    'locations' => [
+      'aws:us-east-2',
+      'aws:eu-central-1',
+      'aws:ca-central-1'
+    ],
+    'options' => {
+      'tick_every' => 60,
+      'min_failure_duration' => 0,
+      'min_location_failed' => 1,
+      'follow_redirects' => true
+    },
+    'message' => 'Test with API',
+    'name' => 'Test with API',
+    'tags' => ['key1:value1', 'key2:value2']
+  }.freeze
+
+  describe '#create_test' do
+    it 'queries the api' do
+      url = api_url + '/synthetics/tests'
+      stub_request(:post, /#{url}/).to_return(body: '{}').then.to_raise(StandardError)
+      expect(dog.send(:create_test, SYNTHETICS_TEST_TYPE, SYNTHETICS_TEST_CONFIG,
+                      SYNTHETICS_TEST_OPTIONS)).to eq ['200', {}]
+
+      expect(WebMock).to have_requested(:post, url)
+    end
+  end
+
+  describe '#edit_test' do
+    it 'queries the api' do
+      url = api_url + "/synthetics/tests/#{SYNTHETICS_TEST_PUBLIC_ID}"
+      stub_request(:put, /#{url}/).to_return(body: '{}').then.to_raise(StandardError)
+      expect(dog.send(:edit_test, SYNTHETICS_TEST_PUBLIC_ID, SYNTHETICS_TEST_TYPE,
+                      SYNTHETICS_TEST_CONFIG, SYNTHETICS_TEST_OPTIONS)).to eq ['200', {}]
+
+      expect(WebMock).to have_requested(:put, url)
+    end
+  end
+
+  describe '#delete_tests' do
+    it_behaves_like 'an api method',
```

Comment:
It seems what we should test against are more complex objects that just a list of ids. See https://docs.datadoghq.com/api/?lang=bash#delete-tests.